typings/mongoengine/fields.pyi

```diff
@@ -1068,3 +1069,87 @@ class ReferenceField(BaseField):
         blank: bool = ...,
     ) -> None: ...
     def __getitem__(self, arg: Any) -> Any: ...
+
+class EnumField(Generic[_ST, _GT], BaseField):
```

Comment:
also could we add a field that uses this in:

https://github.com/sbdchd/mongo-types/blob/3e3a7d6e82ae58cc9d037064c44ef59f831d1800/tests/test_mongoengine.py#L73

as a sort of smoke test